src/subscription_manager/api/__init__.py

```diff
@@ -0,0 +1,48 @@
+#
+# Copyright (c) 2015 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public
+# License as published by the Free Software Foundation; either version
+# 2 of the License (GPLv2) or (at your option) any later version.
+# There is NO WARRANTY for this software, express or implied,
+# including the implied warranties of MERCHANTABILITY,
+# NON-INFRINGEMENT, or FITNESS FOR A PARTICULAR PURPOSE. You should
+# have received a copy of GPLv2 along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+#
+"""
+Class and methods made available in __all__ are provided for use by
+external applications.
+
+All reasonable efforts will be made to maintain compatibility.
+"""
+from functools import wraps
+
+injected = False
+
+
+def request_injection(func):
+    @wraps(func)
+    def func_wrapper(*args, **kwargs):
+        global injected
+        if not injected:
+            from subscription_manager import logutil
+            logutil.init_logger()
```

Comment:
_(comment for discussion, not an issue to be addressed)_

Haven't tested, but this may not be required.

I guess it depends if want any users of our api to also use our logger/handlers/formatters. A standalone script might want to setup it's own logging (say, to stderr). On the other hand, we kind of want all client rhsm'y things to get logged consistently.